src/integration_tests.zig

```diff
@@ -5,119 +5,176 @@ const CairoVM = @import("vm/core.zig").CairoVM;
-        // .{ .pathname = "cairo_programs/fast_ec_add_v2.json", .layout = "all_cairo" },
-        // TODO: FailedToComputeOp0 error
-        // .{ .pathname = "cairo_programs/fast_ec_add_v3.json", .layout = "all_cairo" },
-        // TODO: FailedToComputeOp0 error
-        // .{ .pathname = "cairo_programs/field_arithmetic.json", .layout = "all_cairo" },
-        .{ .pathname = "cairo_programs/fibonacci.json", .layout = "plain" },
-        // TODO: FailedToComputeOp0 error
-        // .{ .pathname = "cairo_programs/field_arithmetic.json", .layout = "all_cairo" },
-        // TODO: merge blake hint
-        // .{ .pathname = "cairo_programs/finalize_blake2s.json", .layout = "all_cairo" },
-        // TODO: merge blake hint
-        // .{ .pathname = "cairo_programs/finalize_blake2s_v2_hint.json", .layout = "all_cairo" },
-        .{ .pathname = "cairo_programs/find_element.json", .layout = "all_cairo" },
-        .{ .pathname = "cairo_programs/fq.json", .layout = "all_cairo" },
-        // TODO: FailedToComputeOp1 error
-        // .{ .pathname = "cairo_programs/fq_test.json", .layout = "all_cairo" },
-        .{ .pathname = "cairo_programs/function_return.json", .layout = "all_cairo" },
-        .{ .pathname = "cairo_programs/function_return_if_print.json", .layout = "all_cairo" },
-        .{ .pathname = "cairo_programs/function_return_to_variable.json", .layout = "all_cairo" },
-        .{ .pathname = "cairo_programs/garaga.json", .layout = "all_cairo" },
-        // TODO: FailedToComputeOp1 error
-        // .{ .pathname = "cairo_programs/highest_bitlen.json", .layout = "all_cairo" },
-        .{ .pathname = "cairo_programs/if_and_prime.json", .layout = "all_cairo" },
-        .{ .pathname = "cairo_programs/if_in_function.json", .layout = "all_cairo" },
-        .{ .pathname = "cairo_programs/if_list.json", .layout = "all_cairo" },
-        // TODO: UnknownOp0 error
-        // .{ .pathname = "cairo_programs/if_reloc_equal.json", .layout = "all_cairo" },
-
-        // .{ .pathname = "cairo_programs/keccak_compiled.json", .layout = "all_cairo" },
-        .{ .pathname = "cairo_programs/keccak_builtin.json", .layout = "all_cairo" },
-        // .{ .pathname = "cairo_programs/keccak_integration_tests.json", .layout = "all_cairo" },
-        // .{ .pathname = "cairo_programs/keccak_copy_inputs.json", .layout = "all_cairo" },
-    };
+    // Refactor: Handle command line arguments
+    const args = try std.process.argsAlloc(allocator);
+    defer std.process.argsFree(allocator, args);
+
+    var test_files_list: ?[]const u8 = null;
+    if (args.len > 1) {
+        // TODO: Parse command-line arguments more rigorously with a CLI utility
+        test_files_list = args[1];
+    }
+
+    var cairo_programs: []CairoTestProgram  = undefined; // This will be defined after reading from the test file or defaulting to all tests.
+
+    // If there are specific tests to run, read them from the file.
+    if (test_files_list) |list| {
```

Comment:
right now this is just a dumb 'if', if there's NO command line arg, just run the programs we define in the source. if there is

- separate by new line
- construct a `CairoTestProgram` where we presume `all_cairo` layout (in the https://github.com/lambdaclass/cairo-vm/blob/main/vm/src/tests/cairo_run_test.rs#L9 file this is the single case where it wasn't using the `all_cairo` layout. need to double check tho as there is loud whispering in the library rn)
- run the programs